packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -329,7 +329,7 @@ export function intOrFloat(value: number): "int32" | "float32" {
  * @param t
  * @returns
  */
-export function isAzureCoreModel(t: Type): boolean {
+export function isAzureCoreTspModel(t: Type): boolean {
```

Comment:
a concern for this helper is that for template model instance, it will also have the namespace of the template, shall we treat them as the azure core model? for example, each time i use `GetResourceOperationStatus` for the following tsp, i will get a concrete `ResourceOperationStatus` type that has `Azure.Core` namespace.
```
@resource("operations")
@parentResource(Resource)
model ResourceOperationStatus<
  Resource extends TypeSpec.Reflection.Model,
  StatusResult = Resource,
  StatusError = Foundations.Error
> is Azure.Core.Foundations.OperationStatus<StatusResult, StatusError>;

@Foundations.Private.ensureVerb("GetResourceOperationStatus", "GET")
  @readsResource(ResourceOperationStatus<Resource>)
  @Foundations.Private.ensureResourceType(Resource)
  GetResourceOperationStatus<
    Resource extends TypeSpec.Reflection.Model,
    StatusResult = Resource,
    StatusError = Foundations.Error,
    Traits extends TypeSpec.Reflection.Model = {}
  > is Foundations.ResourceOperation<
    ResourceOperationStatus<Resource, StatusResult, StatusError>,
    {},
    ResourceOperationStatusResponse<Resource, StatusResult, StatusError>,
    Traits & InterfaceTraits,
    ErrorResponse
  >;
```

Reply:
Since it's taking template namespace into account, wouldn't this mean that `isAzureCoreModel(ResourceOperationStatus) === false`? @timotheeguerin is there a good way to determine whether a model is originally defined in the `Azure.Core` namespace?

Reply:
since typespec team confirm it is by design, let me approve and merge this pr. thanks.